tests/foreman/installer/test_installer.py

```diff
@@ -1337,20 +1362,25 @@
 }
 
 
-def extract_params(lst):
-    """Generator function to extract satellite installer params from lst.
+def extract_help(filter):
+    """Generator function to extract satellite installer params and sections from lines of help text.
     In general lst is cmd.stdout, e.g., a list of strings representing host
     stdout
 
-    :param lst: list  of strings
-    :return: generator with all params
+    :param string filter: Filter `sections` or `params` in full help
+    :return: generator with params or sections depends on filter parameter
     """
-    for line in lst:
-        line = line.strip()
-        first_2_tokens = line.split()[:2]
-        for token in first_2_tokens:
-            if token[0] == '-':
-                yield token.replace(',', '')
+    stdout = ssh.command('satellite-installer --full-help').stdout
+    for line in (stdout or []):
+        if filter == 'sections':
+            line = line.strip()
+            if line.startswith('= '):
+                yield line
+        else:
+            first_2_tokens = line.split()[:2]
```

Comment:
This is no longer calling `strip()` before the split, as that is only done when filter is specified.